File: go/prime_factor/factor_test.go
package prime_factor

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestFactors26(t *testing.T) {
	assert.Equal(t,
		[]int16{0, 0, 0, 0, 2, 0, 2, 0, 2, 3, 2, 0, 2, 0, 2, 3, 2, 0, 2, 0, 2, 3, 2, 0, 2, 5},
		listSmallestPrimeFactors(26),
	)
}

func TestFactors1M(t *testing.T) {
	factors := listSmallestPrimeFactors(1_000_000)
	primes := 1
	for i := 3; i < len(factors); i += 2 {
		if factors[i] == 0 {
			primes++
		}
	}

	assert.Equal(t, 78_498, primes)
}

func TestFactors1B(t *testing.T) {
	factors := listSmallestPrimeFactors(1_000_000_000)
	primes := 1
	for i := 3; i < len(factors); i += 2 {
		if factors[i] == 0 {
			primes++
		}
	}

	assert.Equal(t, 50_847_534, primes)
}

File: python/prime_factor/factor_test.py
from factor import list_smallest_prime_factors


def test_factors_26():
	want = [0, 0, 0, 0, 2, 0, 2, 0, 2, 3, 2, 0, 2, 0, 2, 3, 2, 0, 2, 0, 2, 3, 2, 0, 2, 5]
	assert list_smallest_prime_factors(26) == want


def test_factors_1m():
	factors = list_smallest_prime_factors(1_000_000)
	primes = 1
	for i in range(3, len(factors), 2):
		if factors[i] == 0:
			primes += 1

	assert primes == 78_498


def test_factors_10m():
	factors = list_smallest_prime_factors(10_000_000)
	primes = 1
	for i in range(3, len(factors), 2):
		if factors[i] == 0:
			primes += 1

	assert primes == 664_579
